openedx/openedx-platform
# feat: add course optimizer waffle flag

## Description

Add a waffle flag to toggle on or off the new Course Optimizer feature per course (https://2u-internal.atlassian.net/browse/TNL-11808).

#### Impact

Course Author

#### Context

Course Optimizer is a much-requested feature that has been aligned with the OpenEdx community. The only component for now is a broken link checker that will return a list of all broken links in a course to the frontend. There will be a new page for this that is a bit similar to the export page.

This PR only deals with adding a waffle flag.

## Supporting information

Internal Ticket (2U):
https://2u-internal.atlassian.net/browse/TNL-11808

I will add some other links to related issues and documentation soon.

## Testing instructions

- Go to django admin -> waffle flags
- Add a new waffle flag "contentstore.enable_course_optimizer)" and set "everyone" to "Yes"
- Make sure that the waffle flag "contentstore.new_studio_mfe.use_new_updates_page" is also set to "Yes" for everyone
- Go to studio -> open a course -> Open the course updates page at http://localhost:2001/course/course-v1:edX+E202+1T2024/course_info
- Go to network tab and search for a request to `
http://localhost:18010/api/contentstore/v1/course_waffle_flags/[...]`
- The response should include `"enable_course_optimizer": true`
- Go back to django admin and set the flag "contentstore.enable_course_optimizer" for everyone to "No"
- Go back to the studio course, reload, and check that the response should say `"enable_course_optimizer": false`

## Deadline

TBD
